lib/farscape/transition.rb

```diff
@@ -48,6 +48,13 @@ def invoke(args = {})
         # This makes it impossible to use URIs with parameters.
         call_options[:url] = @transition.uri
         call_options[:body] = (call_options[:body] || {}).merge(params) if params.present?
+
+        if call_options[:headers][:Accept] =~ /\Aapplication\/.*json\z/ &&
+          call_options[:body] && call_options[:body].respond_to?(:to_json)
```

Comment:
- the multiple `&&` should be one per line
- `call_options[:body]` should be the first condition so we don't do needless pattern matching if there's no body
- that regex should use something like `%r(...)` so you don't need to escape the slash
- the accept header can have many values so `\A` and `\z` shouldn't be here I think (see [wikipedia](https://en.wikipedia.org/wiki/Content_negotiation))